packages/netlify-cms-widget-list/src/ListControl.js

```diff
@@ -321,11 +321,14 @@ export default class ListControl extends React.Component {
     const { field } = this.props;
     if (this.getValueType() === valueTypes.MIXED) {
       const itemType = getTypedFieldForValue(field, item);
-      const fallbackLabel = itemType.get('label');
-      const summaryTemplate = itemType.get('summary');
-      return summaryTemplate
-        ? compileStringTemplate(summaryTemplate, null, '', item)
-        : fallbackLabel;
+      const label = itemType.get('label', field.get('name'));
+      const summary = itemType.get('summary');
+      let labelReturn = label
+      if (summary) {
+        const data = item.set('label', label)
+        labelReturn = stringTemplate.compileStringTemplate(summary, null, '', data)
```

Comment:
@erezrokah It looks like in a string template, `slug` and date-related keys are hard-coded in:

https://github.com/netlify/netlify-cms/blob/0ac3592e2617ef2c3b108d4256cd977787dcf4b2/packages/netlify-cms-lib-widgets/src/stringTemplate.ts#L98-L104

When I tried using `{{fields.label}}`, it returns the same value as `{{label}}`. Should I pass in type's `label` under a different name/map?

Reply:
The `{{fields.*}}` format is useful when you want to avoid collision with built in tags, for example let's say you want to reference a field named `slug` (see note here https://www.netlifycms.org/docs/configuration-options/#slug).
When referencing a field I would say it is safer to always use `{{fields.fieldName}}` to avoid possible future collisions (for example we recently added built in `filename` and `extension` template tags)

Reply:
Hi @erezrokah, thank you for further explaining about the `field.` format. I think I might be missing something, let me walk through this issue I have in my head:

Say I have an `item` Map with `{ label: 'One' }`

The following code will replace `label` with `item.getIn(['label'])`, returning `'Type: One'`.

```js
compileStringTemplate("Type: {{label}}", null, null, item) // "Type: One"
``` 

https://github.com/netlify/netlify-cms/blob/0ac3592e2617ef2c3b108d4256cd977787dcf4b2/packages/netlify-cms-lib-widgets/src/stringTemplate.ts#L106

I can override this `label` field by adding a new field to `item`, and now the following code should return `Type: Two`:

```js
item = item.set('label', 'Two')
compileStringTemplate("Type: {{label}}", null, null, item) // "Type: Two"
```
This works!

---
Now here's the part I'm confused about. The following code should replace `fields.label` with the original `label` in `item` & return `Type: One`. 

```js
compileStringTemplate("Type: {{fields.label}}", null, null, item) // "Type: Two"
```

This doesn't work.

Looking at the code of `getExplicitFieldReplacement` which handles keys with `fields.` prefix, it removes `fields.` prefix and use `data.getIn(keyToPathArray(fieldName))` to access the value of the `label`.

https://github.com/netlify/netlify-cms/blob/0ac3592e2617ef2c3b108d4256cd977787dcf4b2/packages/netlify-cms-lib-widgets/src/stringTemplate.ts#L69-L70

Looking at the code of `compileStringTemplate`, if a field is not a date-related key (`year`, `month`, `day`, etc), doesn't start with `fields.`, or is not `slug`, it will access the value of the key via `data.getIn(keyToPathArray(key), '')`.

https://github.com/netlify/netlify-cms/blob/0ac3592e2617ef2c3b108d4256cd977787dcf4b2/packages/netlify-cms-lib-widgets/src/stringTemplate.ts#L96-L107

Because of this, `compileStringTemplate` does exact the same for `fields.label` and `label` in stringTemplate:

```js
  let data = Map({ label: 'One' })
  data = item.set('label', 'Two')
  labelReturn = stringTemplate.compileStringTemplate('{{fields.label}} | {{label}}', null, '', data)
```

I see no special handling for `filename` and `extension` either, so I'm lost on understanding how it works. Do I have to do something special so the original data is preserved? Thank you!

Reply:
Sorry for not being clear enough. At the moment `{{label}}` `{{fields.label}}` have the same result, but we might want to have a built in `label` with special meaning in the future (like `date` and `slug`). If we do decide to make `label` a built in template var it would break behaviour for existing users that use `{{label}}` without the `fields` prefix and we would need to apply backwards compatibility code for that. To avoid doing so in the widget as well using `fields.label` is safer.
As for `filename` and `extension`, they are added when relevant here https://github.com/netlify/netlify-cms/blob/1c7ef1c457612242257d96af53a57c1460e45b31/packages/netlify-cms-core/src/lib/formatters.ts#L206
For example, adding those templates when evaluating the slug doesn't make any sense since the filename is based on the slug.